### PR TITLE
Get rid of fiber_set_cancellable in 4 more places

### PR DESCRIPTION
**Pull request by Gumix**
See commit messages.

Part of #7166

### CODE REVIEW

File: src/box/recovery.cc [hunk 1]
```diff
@@ -500,14 +499,10 @@ hot_standby_f(va_list ap)
...
-		if (subscription.events == 0) {
-			/**
-			 * Allow an immediate wakeup/break loop
-			 * from recovery_stop_local().
-			 */
-			fiber_set_cancellable(true);
+		while (r->watcher != NULL &&
+		       !timed_out &&
+		       subscription.events == 0) {
 			timed_out = fiber_yield_timeout(wal_dir_rescan_delay);
```
[the remark this answers is not in the document]
Comment (Gumix) [reply src/box/recovery.cc]: Done.

File: test/unit/cord.c [hunk 2]
```diff
@@ -0,0 +1,69 @@
...
+int
+main(void)
+{
+	header();
+	plan(2);
+
+	memory_init();
+	fiber_init(fiber_c_invoke);
+
+	main_fiber = fiber_new("main", main_fn);
```
[the remark this answers is not in the document]
Comment (Gumix) [reply test/unit/cord.c]: Done.

**Comment by locker** [comment]
@Gumix Please fix the compilation error on FreeBSD.

**Comment by Gumix** [comment]
> @Gumix Please fix the compilation error on FreeBSD.

Fixed.

**Pull request merged**